code/modules/mob/emote.dm

```diff
@@ -77,6 +77,9 @@
 
 		// Type 1 (Visual) emotes are sent to anyone in view of the item
 		if(m_type & EMOTE_VISUAL)
+			var/runechat_text = input
+			if(length(input) > 100)
+				runechat_text = "[copytext(input, 1, 101)]..."
```

Comment:
There seems to be a system for this already in `/datum/chatmessage/proc/generate_image()` which shortens regular say messages. The only reason it's not doing the same for emotes is because they don't have a span class assigned to them, so `split_html()` doesn't return an `inner_text` value.
https://github.com/ParadiseSS13/Paradise/blob/2f5951bbaa2ddc94696c0f4a04b7bda70f232fc8/code/datums/chatmessage.dm#L100-L110
It might be better to use this existing system rather than making a new one for emotes, assuming that wouldn't end up being too convoluted.